pkg/csi-handler/pv_checker.go

```diff
@@ -149,14 +161,18 @@ func (checker *PVHealthConditionChecker) CheckControllerVolumeStatus(pv *v1.Pers
 		return err
 	}
 
+	pvc, err := checker.pvcLister.PersistentVolumeClaims(pv.Spec.ClaimRef.Namespace).Get(pv.Spec.ClaimRef.Name)
+	if err != nil {
+		return err
+	}
+
 	// At the first stage, we just send PVC events
 	if volumeCondition.GetAbnormal() {
 		// Since pv status is bound, we believe PV controller, do not check pv.Spec.ClaimRef here.
-		pvc, err := checker.pvcLister.PersistentVolumeClaims(pv.Spec.ClaimRef.Namespace).Get(pv.Spec.ClaimRef.Name)
-		if err != nil {
-			return err
-		}
 		checker.eventRecorder.Event(pvc, v1.EventTypeWarning, "VolumeConditionAbnormal", volumeCondition.GetMessage())
+	} else {
```

Comment:
What if error occurs when we execute API call ?
Default `abnormal` is True or False ?  IIUC, it is False
So we cann't directly send recovery events here.

Reply:
You mean `ControllerGetVolumeCondition` API was invoked failed. Right?
In that case, It should return from [here](https://github.com/kubernetes-csi/external-health-monitor/pull/36/files#diff-31795038ad9c7264079b907f4a04a702R161).

Reply:
Yes, right, i missed that part, thanks